src/java/org/apache/cassandra/index/SecondaryIndexManager.java

```diff
@@ -491,9 +497,83 @@ public static String getIndexName(String cfName)
         return StringUtils.substringAfter(cfName, Directories.SECONDARY_INDEX_NAME_SEPARATOR);
     }
 
+    // TODO: JavaDoc if this approach survives...mention that this should not be used for live SSTables
+    public void validateStorageAttachedIndexes(Collection<SSTableReader> sstables) throws IOException
```

Comment:
Right now, this method exists solely to validate SAI components streamed entirely before we allow a streaming transaction to commit. It therefore replaces any such validation that would occur via notification in `StorageAttachedIndexGroup` (which should only worry about adding the new SSTable indexes to the view, etc.).

This might also be useful for import, however, in the case where we bring in new SSTables that already have built index components. We'd just have to make sure we only did it in this case and not when we need to build them anyway...

Reply:
I think it would be possible to just add an argument indicating a failure mode to this method. Streaming could fail hard if missing anything in the entire SSTable case, and import could just validate what's there, given the rest will be built anyway immediately after.

Reply:
This is what I ended up doing.

i.e. `boolean validateStorageAttachedIndexes(Collection<SSTableReader> sstables, boolean failIncomplete) throws IOException`